Assets/Scripts/Dungeon/Room Spawn.cs

```diff
@@ -0,0 +1,87 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class RoomSpawn : MonoBehaviour
+{
+    //private RoomTemplates template;
+    private RoomTemplates roomTemplates;
+    private checkroomcollision checkroom;
+    public GameObject deadEnd;
+    public GameObject template;
+
+    public bool collide;
+    public int attempts;
+
+    //public List<GameObject> Rooms;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        attempts = 0;
+        spawnRoom();
+    }
+
+    public bool collideTrue(bool RoomCollide)
+    {
+        collide = RoomCollide;
+        //Debug.Log("collide: " + collide);
```

Comment:
Getting rid of debug messages will be good to not clutter the console in the future. When everyone makes debug branches and non of them relate to what youre working on it'll be chaotic and take time to go through deleting in the future. that was just a though on my end but this goes with deleteing commented code you dont use like I was sayng in another comment. Good for clean code and readability.